pages/api/sigil/[address]/tokenId/[tokenId].ts

```diff
@@ -0,0 +1,263 @@
+//The Structs//
+///////////////
+
+type ProcessedTokenData = {
+  hash: string
+  phrase_value: string
+  binary_value: string
+  type: hashType
+};
+
+type ProcessedWalletData = {
+  dao: number
+  non_dao: number
+  votes: number
+  proposals: number
+};
+
+/////////////////////////////////
+//The Data Collection Functions//
+/////////////////////////////////
+
+//Gets the Token Hashes data from pre-existing API
+async function getTokenAPIData(tokenId: number) {
+  try {
+    const APIdata = await fetch(`http://permutation-3.vercel.app/api/tokenId/${tokenId}`);
+
+    const data = await APIdata.json();
+
+    return data;
+
+  } catch (error) {
+    console.error(`error calling API for tokenID data: ${error}`);
+  }
+}
+
+//Returns processed data for easy use when constucting the NFT
+function processTokenAPIData(data: any): ProcessedTokenData {
+
+  const processedTokenData = {} as ProcessedTokenData;
+
+  try {
+
+    processedTokenData.hash = data.hash;
+
+    processedTokenData.phrase_value = data.phrase_value;
+
+    processedTokenData.binary_value = data.binary_value;
+
+    processedTokenData.type = data.type;
+
+  } catch (error) {
+    console.error(`error transposing token data: ${error}`);
+  }
+
+  return processedTokenData;
+}
+
+//Gets the Wallet Hashes data from pre-existing API
+async function getWalletAPIData(address: string) {
+  try {
+
+    const APIdata = await fetch(`http://permutation-3.vercel.app/api/wallet/hashes/${address}`);
+
+    const data = await APIdata.json();
+
+    return data;
+
+  } catch (error) {
+    console.error(`error calling API for Wallet data: ${error}`);
+  }
+}
+
+//Returns processed data for easy use when constucting the NFT
+function processWalletAPIData(data: any): ProcessedWalletData {
+
+  const processedWalletData = {} as ProcessedWalletData;
+
+  try {
+    //WalletData will return an undefined if the owner does not own any Hashes NFTs
+    if (data === undefined) {
+
+      processedWalletData.dao = 0;
+      processedWalletData.non_dao = 0;
+      processedWalletData.votes = 0;
+      processedWalletData.proposals = 0;
+    }
+    else {
+
+      var dao = 0;
+      var non_dao = 0;
+
+      for (let i = 0; i < data.hashes.length; i++) {
+
+        //Adds entry to either DAO or non-DAO increment
+        if (data.hashes[i].type === "DAO") {
+          dao += 1;
+        }
+        else {
+          non_dao += 1;
+        }
+      }
+
+      processedWalletData.dao = dao;
+      processedWalletData.non_dao = non_dao;
+      processedWalletData.votes = data.on_chain_votes;
+
+      if (data.proposals_created === undefined) {
+        processedWalletData.proposals = 0;
+      } else {
+        processedWalletData.proposals = data.proposals_created;
+      }
+    }
+  } catch (error) {
+    console.error(`error transposing wallet data: ${error}`);
+  }
+
+  return processedWalletData;
+}
+
+//////////////////////////////////////////////
+//Helper Functions For Each Sigil Collection//
```

Comment:
Your choice re: comment style here; a good practice is to only use comments when things are very hazy, when the code cannot speak for itself, and if something has been created that has to do with a specific outlier case